### ISSUE TITLE
Scoping iframe by non-ID selectors

### ISSUE THREAD

**Issue by almostwhitehat** [natritmeyer/site_prism#118]
If I am reading http://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FSelenium%2FDriver%3Awithin_frame correctly, it looks like Capybara is now able to scope an iframe by an actual element, instead of just an ID. Would it be possible for Site Prism to now allow mapping iframes by a more generic selector than just ID?

**Comment by liquid1982** [comment]
I ran into the same problem – it's currently not possible to refer to an iframe if it doesn't have an`id` attribute set. Here's the ugly workaround I came up with.

```ruby
class MyIFrame < SitePrism::Page
end

class MyPage < SitePrism::Page
  set_url 'http://example.com'

  iframe :my_iframe, MyIFrame, '.iframe-container #my-iframe'

  def add_id_attribute_to_iframe
    page.evaluate_script("document.querySelector('.iframe-container iframe').id = 'my-iframe'")
  end
end

my_page = MyPage.new.tap(&:load)
my_page.add_id_attribute_to_iframe
my_page.my_iframe do |iframe|
  expect(iframe).to have_text 'My Text'
end
```

**Comment by ditchfieldcaleb** [comment]
Any updates on this?

**Comment by luke-hill** [comment]
Hi there. I'm going to be looking into the entire history list of issues e.t.c. and this one appears to be stale. Could you give a brief update as to what you plan to do with this.

If you plan to land a PR soon or already have done. Can you close this issue
If you think it's no longer relevant. Can you close this issue.

If possibly can you provide some update in the next couple of weeks (I'm aware this issue is several years old)

**Comment by luke-hill** [comment]
Have re-looked into this and this seems to be something we should support / currently a bug.

**Comment by luke-hill** [comment]
Closing issue. Feature will be released in 2.12 of the gem
